lazy_dataset/core.py

```diff
@@ -338,6 +338,18 @@ def indexable(self) -> bool:
 
     @property
     def ordered(self) -> bool:
+        """
+        Indicate whether each iteration over the dataset yield the examples
+        in the same order.
+
+        This property is used for the `cache` implementation.
```

Comment:
Is this comment necessary? If we add something that needs `ordered` and is not caching, we will certainly not remove this comment.

Reply:
Recently when I looked in the source code, I was unsure, where we needed this property, because it indicates a special property.
After checking, I only found the cache dataset that needs this property as of now.
Therefore, I thought it would be useful to document the use case for those that implement custom datasets.